### PR TITLE
IGAPP-293: Setup intelliJ project(s)

### PR DESCRIPTION
**Pull request by maxammann**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/.circleci [hunk 1]
```diff
@@ -0,0 +1 @@
+../.circleci
```
Comment (klinzo) [inline native/.circleci:1]: Does this work on windows?

File: native/android/.circleci [hunk 2]
```diff
@@ -0,0 +1 @@
+../.circleci
```
Comment (klinzo) [inline native/android/.circleci:1]: Why do we need this here as well?

**Review by klinzo — APPROVED** [review]
I think symlinks to README, version.json, package.json, root/docs and root/tools would be nice as well :)

This way it is not possible anymore to open the root project in intellij, is it?

File: native/android/.circleci (hunk 2 above)
Comment (maxammann) [reply native/android/.circleci:1]: I thought I'd add this here as you can have the integreat-app-android project open and nothing else. I kind of added it for consistency. But I agree that we could remove it.

File: native/.circleci (hunk 1 above)
Comment (maxammann) [reply native/.circleci:1]: :shrug: 
Probably not :D

Comment (Taggotty) [reply native/.circleci:1]: Does not work on windows...

File: native/.idea/codeStyles/Project.xml [hunk 3]
```diff
@@ -13,6 +13,24 @@
...
+          <package name="io.ktor" alias="false" withSubpackages="true" />
+        </value>
+      </option>
+      <option name="PACKAGES_IMPORT_LAYOUT">
+        <value>
+          <package name="" alias="false" withSubpackages="true" />
+          <package name="java" alias="false" withSubpackages="true" />
+          <package name="javax" alias="false" withSubpackages="true" />
+          <package name="kotlin" alias="false" withSubpackages="true" />
+          <package name="" alias="true" withSubpackages="true" />
```
Comment (Taggotty) [inline native/.idea/codeStyles/Project.xml:30]: why are there 2 packages `""` with different configurations?

**Review by Taggotty — APPROVED** [review]

File: native/.circleci (hunk 1 above)
Comment (Taggotty) [reply native/.circleci:1]: On windows there should also be symlinks, but they seem to work slightly different. Tried using a basic link, but IntelliJ doesn't know what to do with that. I only found that you could use content roots in intellij, but it doesn't really look that fancy, so I would not recommend them.

**Comment by klinzo** [comment]
What is the problem with opening the whole project at once by the way? I like it personally and the performance is reasonable fast (not noticeable slower at least). More overview over the whole project, no need to have multiple windows to compare stuff and no symlinks needed :thinking:

**Comment by Taggotty** [comment]
> What is the problem with opening the whole project at once by the way? I like it personally and the performance is reasonable fast (not noticeable slower at least). More overview over the whole project, no need to have multiple windows to compare stuff and no symlinks needed :thinking: 

Personally my biggest concern is that intellij takes quite some time to Index the Projekts, especially when switch Ing between Branche. Currebtly native and Web are excluded folder So they are not indexed. But if we would add then e. g. as content root that would definitely change.

**Comment by maxammann** [comment]
I think it does not make sense to allow this because of flow, eslint etc.

**Comment by maxammann** [comment]
We would need to merge the versions etc

**Pull request merged**